src/workloads/docs/SamplingLoader.yml

```diff
@@ -0,0 +1,112 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload demonstrates using the SamplingLoader to expand the contents of a collection. This
+  actor will take a random sample of a configurable size and re-insert the same documents. It will
+  project out the "_id" field before re-inserting to avoid duplicate key errors.
+GlobalDefaults:
+  db: &db test
+  Collection: &Collection sampling_loader_demo
+
+Actors:
+# This is just to seed the collection with 40 documents so that we'll have some sample data to draw
+# from.
+- Name: CollectionSeed
+  Type: MonotonicSingleLoader
+  Phases:
+  - Collection: *Collection
+    Database: *db
+    Repeat: 1
+    Document:
+      x: {^Inc: {}}
+    DocumentCount: 40
+    BatchSize: 10
+  - {Nop: true}
+  - {Nop: true}
+  - {Nop: true}
+
+# Our first demonstration of the sampling loader.
+- Name: ImpliedSampleSizeDemo
+  Type: SamplingLoader
+  Threads: 1
+  Phases:
+  - {Nop: true}
+  - Repeat: 1
+    Database: *db
+    Collection: *Collection
+    # Number of documents in a single insert operation.
+    InsertBatchSize: 5
+    # Number of insert operations. The total number of documents to be inserted is
+    # Batches * InsertBatchSize.
+    Batches: 2
+    # Number of documents in the sample. The loader will cycle through the sampled docs, using them
+    # as templates # for the ones to be incerted, until it inserts the requested number of batches.
+    # For example, if the sample size is set to 1, exactly one document is sampled from the original
+    # dataset and reused as a template for  _all_ of the inserted docs. And if the sample size is
+    # equal to the insert batch size, each sampled document will be "re-inserted" as many times as
+    # the number of batches.
+    #
+    # This may be useful if you want to insert a lot of documents to measure that performance, but
+    # don't necessarily need a big sample and want to benefit from the random cursor optimization for
+    # small sample sizes to keep the test's runtime lower.
+    #
+    # Note that the sample size does not need to be correlated or a multiple of the insert batch size
```

Comment:
This is a good point. Maybe set SampleSize in the example to 7 or something like that to highlight it? The effect of `SampleSize: 1` is described above and is more "intuitive".

Reply:
Good idea, done!